js/src/redux/reducers/snippetEditor.js

```diff
@@ -40,17 +41,37 @@ function snippetEditorReducer( state = INITIAL_STATE, action ) {
 				},
 			};
 
-		case UPDATE_REPLACEMENT_VARIABLE:
-			return {
-				...state,
-				replacementVariables: [
-					...state.replacementVariables,
-					{
+		case UPDATE_REPLACEMENT_VARIABLE: {
+			const index = state.replacementVariables.findIndex( replaceVar => {
```

Comment:
This could also be done in the map loop by having a boolean if it was found or not. Then if it was not found add it to the end. Like so:

```
let append = true;
let newReplacementVariables = state.replacementVariables.map( ( replaceVar ) => {
	if ( replaceVar.name === action.name ) {
		append = false;
		return {
			name: action.name,
			value: action.value,
		};
	}
	return replaceVar;
} );
if ( append ) {
	newReplacementVariables.push( {
		name: action.name,
		value: action.value,
	} );
}
```